spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectionFactory.scala

```diff
@@ -73,9 +96,36 @@ object DefaultConnectionFactory extends CassandraConnectionFactory {
       .withSocketOptions(options)
   }
 
+  /** Returns the Cluster.Builder object used to setup Cluster instance. */
+  def clusterSSLBuilder(conf: CassandraConnectorConf, builder: Cluster.Builder): Cluster.Builder = conf.sslEnabled match {
+    case false => builder
+    case true => clusterSSLBuilderOptions(conf, builder)
+  }
+
+  private def clusterSSLBuilderOptions(conf: CassandraConnectorConf, builder: Cluster.Builder): Cluster.Builder = conf.sslTrustStorePath match {
```

Comment:
Would you refactor this class a little bit? I would like the cluster build created in one place - `clusterBuilder` method. Why don't you make a method like `maybeCreateSSLOptions` and optionally apply it inside `clusterBuilder` method? I would keep `createCluster` method just calling `clusterBuilder(conf).build()` like it was before.